drake/systems/plants/RigidBody.h

```diff
@@ -56,17 +56,17 @@ class DRAKERBM_EXPORT RigidBody {
     this->collision_filter_ignores &= ~group;
   };
 
-  bool adjacentTo(const std::shared_ptr<RigidBody>& other) const {
-    return ((parent == other && !(joint && joint->isFloating())) ||
-            (other->parent.get() == this &&
-             !(other->joint && other->joint->isFloating())));
+  bool adjacentTo(const RigidBody& other) const {
+    return ((parent.get() == &other && !(joint && joint->isFloating())) ||
```

Comment:
This and some of the nearby methods would be clearer with a utility method something like
```c++
return (parent->isSameRigidBody(other) && ...);
```